src/util.ts

```diff
@@ -38,3 +38,32 @@ export function safeJsonParse(jsonString: string): ParsedSgidDataValue {
     return jsonString
   }
 }
+
+function isDefinedObject(value: unknown): value is Record<string, unknown> {
+  return (
+    typeof value === 'object' &&
+    value !== null &&
+    value !== undefined &&
+    !Array.isArray(value)
+  )
+}
+
+export function isSgidUserinfoObject(
+  data: unknown,
+): data is Record<string, string> {
+  if (!isDefinedObject(data)) {
+    return false
+  }
+
+  let result = true
```

Comment:
nit: can consider this code instead of line 58-68 for code readability
```suggestion
  return Object.values(data).every((value) => typeof value === 'string')
```

Reply:
thanks, fixed in [b2060](https://github.com/opengovsg/sgid-client/pull/69/commits/b206067be2e7bf1841660a5561e75a757b2a711e)

btw for future suggested changes, you would need to drag to include the entire section that you want to replace - when i committed your suggestion directly, there was leftover code that needed to be removed